src/Makefile.am

```diff
@@ -33,8 +32,15 @@ if ENABLE_BINRELOC
 geany_LDFLAGS += -Wl,-rpath='$$ORIGIN/../lib'
 endif
 
+TM_HDRS = \
+	tagmanager/tm_source_file.h \
+	tagmanager/tm_tag.h \
+	tagmanager/tm_workspace.h \
+	tagmanager/tm_parser.h
+
 geany_includedir = $(includedir)/geany
 geany_include_HEADERS = \
+	$(TM_HDRS) \
```

Comment:
You want to create a separate variable like this:

tagmanager_includedir = $(includedir)/geany/tagmanager
tagmanager_include_HEADERS = $(TM_HDRS)

Then headers will be installed in the subdirectory.